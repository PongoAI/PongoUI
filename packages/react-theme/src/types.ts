import type { ColorSet } from '@pongo-ui/react-theme-generator';

export type Theme = ColorTokens & Fonts;

/**
 * Color theme tokens.
 */
export type ColorTokens = { textColor: string; canvasColor: string } & ColorSet<
  'inherit' | 'brand' | 'secondary' | 'success' | 'danger' | 'social' | 'warning' | 'info'
>;

export type Fonts = FontFamilyTokens & FontOpacityTokens & FontSizeTokens & FontLineHeightTokens & FontWeightTokens;

/**
 * Font family tokens.
 */
export type FontFamilyTokens = {
  baseFont: string;
  monospaceFont: string;
};

/**
 * Font opacity tokens
 */
export type FontOpacityTokens = {
  display: string;
  subHeadline: string;
  body: string;
  caption: string;
  title: string;
  header1: string;
  header2: string;
  header3: string;
};

/**
 * Font size tokens.
 */
export type FontSizeTokens = {
  fontSize100: string;
  fontSize200: string;
  fontSize300: string;
  fontSize400: string;
  fontSize500: string;
  fontSize600: string;
  fontSize700: string;
  fontSize800: string;
  fontSize900: string;
  fontSize1000: string;
};

/**
 * Font line height tokens.
 */
export type FontLineHeightTokens = {
  lineHeight100: string;
  lineHeight200: string;
  lineHeight300: string;
  lineHeight400: string;
  lineHeight500: string;
  lineHeight600: string;
  lineHeight700: string;
  lineHeight800: string;
  lineHeight900: string;
  lineHeight1000: string;
};

/**
 * Font weight tokens.
 */
export type FontWeightTokens = {
  thin: number;
  light: number;
  regular: number;
  medium: number;
  bold: number;
  black: number;
};
