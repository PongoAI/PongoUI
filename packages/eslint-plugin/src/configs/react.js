/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
