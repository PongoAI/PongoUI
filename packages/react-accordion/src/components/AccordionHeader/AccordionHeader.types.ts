import type {
  AccordionHeaderProps as FluentAccordionHeaderProps,
  AccordionHeaderState as FluentAccordionHeaderState,
  AccordionHeaderSlots as FluentAccordionHeaderSlots,
} from '@fluentui/react-accordion';

export type AccordionHeaderSlots = FluentAccordionHeaderSlots;

export type AccordionHeaderProps = FluentAccordionHeaderProps;

export type AccordionHeaderState = FluentAccordionHeaderState;
