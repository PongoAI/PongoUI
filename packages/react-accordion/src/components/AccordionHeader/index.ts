export * from './AccordionItem';
export * from './AccordionItem.types';
