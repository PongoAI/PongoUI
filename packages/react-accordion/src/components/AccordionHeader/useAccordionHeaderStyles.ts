import { shorthands, makeStyles, mergeClasses } from '@griffel/react';
import { tokens } from '@pongo-ui/react-theme';
import type { AccordionHeaderState } from './AccordionHeader.types';

const useStyles = makeStyles({
  root: {
    position: 'relative',
    fontFamily: tokens.fontFamilyBase,
    fontSize: tokens.fontSize300,
  },

  enabled: {
    cursor: 'pointer',
    color: tokens.textColor,
  },

  disabled: {
    cursor: 'not-allowed',
    color: tokens.inheritDisabled,
  },

  expandIcon: {},
});

export const useAccordionHeaderStyles = (state: AccordionHeaderState) => {
  const styles = useStyles();

  state.root.className = mergeClasses(
    styles.root,
    state.disabled ? styles.disabled : styles.enabled,
    state.root.className,
  );

  state.button.className = mergeClasses(state.disabled ? styles.disabled : styles.enabled, state.button.className);

  if (state.expandIcon) {
    state.expandIcon.className = mergeClasses(styles.expandIcon, state.button.className);
  }

  return state;
};
