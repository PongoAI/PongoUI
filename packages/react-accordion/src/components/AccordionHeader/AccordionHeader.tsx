import * as React from 'react';
import {
  useAccordionHeaderContextValues_unstable,
  useAccordionHeader_unstable,
  renderAccordionHeader_unstable,
} from '@fluentui/react-accordion';
import { useAccordionHeaderStyles } from './useAccordionHeaderStyles';
import type { AccordionHeaderProps } from './AccordionHeader.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The AccordionItem component is used to show and hide nested information.
 */
export const AccordionHeader: ForwardRefComponent<AccordionHeaderProps> = React.forwardRef((props, ref) => {
  const state = useAccordionHeader_unstable(props, ref);

  const contextValues = useAccordionHeaderContextValues_unstable(state);

  useAccordionHeaderStyles(state);

  return renderAccordionHeader_unstable(state, contextValues);
});

AccordionHeader.displayName = 'AccordionHeader';
