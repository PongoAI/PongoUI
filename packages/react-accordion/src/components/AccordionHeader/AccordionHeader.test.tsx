import * as React from 'react';
import { AccordionHeader } from './AccordionHeader';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';
import { AccordionHeaderProps } from './AccordionHeader.types';

describe('AccordionHeader', () => {
  isConformant({
    Component: AccordionHeader as React.FunctionComponent<AccordionHeaderProps>,
    displayName: 'AccordionHeader',
    disabledTests: ['component-handles-classname'],
  });

  describe('Snapshot tests', () => {
    it('renders a basic AccordionHeader', () => {
      const result = render(<AccordionHeader>Test</AccordionHeader>);

      expect(result.baseElement).toMatchSnapshot();
    });
  });
});
