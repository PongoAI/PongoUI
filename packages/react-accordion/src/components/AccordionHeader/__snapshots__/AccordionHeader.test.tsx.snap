// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionHeader Snapshot tests renders a basic AccordionHeader 1`] = `
<body>
  <div>
    <div
      class="___1qjmscb fvalcr8 f1wac4k f1t8un4 f1qo5dib f10pi13n f22iagw f122n59 f1p3nwhy f11589ue f1q5o8ev f1pdflbu f3rmtva f1e4lqlz fc467gg f1mo0ibp f1arxnym fytdu2e f37px4s f1xqy1su fgusgyc f1k6fduh fvfw249"
    >
      <button
        aria-expanded="false"
        class="___pdfdv00 f10pi13n f22iagw f122n59 f1p3nwhy f11589ue f1q5o8ev f1pdflbu f3rmtva f1e4lqlz fc467gg f1mo0ibp f1arxnym fytdu2e f37px4s f1xqy1su fgusgyc f1k6fduh fvfw249"
      >
        <span
          aria-hidden="true"
        >
          <svg
            aria-hidden="true"
            class="___1yjlmar f1w7gpdv fez10in f1kxrtqx"
            fill="currentColor"
            height="1em"
            transform="rotate(0)"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.65 4.15c.2-.2.5-.2.7 0l5.49 5.46c.21.22.21.57 0 .78l-5.49 5.46a.5.5 0 01-.7-.7L12.8 10 7.65 4.85a.5.5 0 010-.7z"
            />
          </svg>
        </span>
        Test
      </button>
    </div>
  </div>
</body>
`;
